enhancements/dns/externaldns-operator-aws-assume-role.md

```diff
@@ -0,0 +1,301 @@
+
+## Design Details
+
+### Open Questions
+
+N/A
+
+### Test Plan
+
+We will add E2E tests for the ExternalDNS Operator utilizing two AWS CI accounts to create DNS records in one (Account
+A) and install the cluster in the other (Account B). This E2E test will leverage existing steps in the step registry
+as created by [AWS Shared VPC with Cross-account DNS Zones](/enhancements/installer/aws-cross-account-dns-zone.md) such
+as [ipi-aws-pre-shared-vpc-phz](https://steps.ci.openshift.org/chain/ipi-aws-pre-shared-vpc-phz) and [ipi-aws-post-shared-vpc-phz](https://steps.ci.openshift.org/chain/ipi-aws-post-shared-vpc-phz).
+These steps set up a shared VPC cluster with access to a cross-account DNS private hosted zone. However, we may choose to use
+a public hosted zone instead, as there is no functional distinction between the two, and it simplifies the E2E testing
+process.
+
+### Graduation Criteria
+
+This update is an extension of the existing v1beta1 ExternalDNS Operator API and does not impact the graduation
+criteria for the v1beta1 API.
+
+The ExternalDNS Operator updates to support Shared VPC are out of payload and are therefore not aligned with
+OpenShift releases. Initially, the plan was to release this enhancement alongside of OCP 4.14. However, because
+ExternalDNS Operator lacks STS support (see [Lack of STS Support](#lack-of-sts-support)), the decision was made to
+complete implementation of this feature, but postpone the release until STS support is available.
+
+There is no requirement for backporting this feature, as outlined in [AWS Shared VPC with Cross-account DNS Zones](/enhancements/installer/aws-cross-account-dns-zone.md).
+
+#### Dev Preview -> Tech Preview
+
+Updates to the ExternalDNS Operator will be made available without being restricted by a feature gate. These updates
+will be considered GA at the time of merging the feature.
+
+#### Tech Preview -> GA
+
+See [Dev Preview -> Tech Preview](#dev-preview---tech-preview).
+
+#### Removing a deprecated feature
+
+N/A
+
+### Upgrade / Downgrade Strategy
+
+* As outlined in [The Required Credentials Issue](#the-required-credentials-issue), when we bump the API from v1beta1,
+  the conversion webhook will be required to handle the changes we make to the `Credentials` field. Nonetheless, in the
+  initial implementation of this API, we will have no concern related to upgrading or downgrading.
+
+### Version Skew Strategy
+
+The updates to the ExternalDNS Operator are decoupled from the standard release schedule of OpenShift Container
+Platform. Consequently, in a shared VPC cluster using cross-account DNS record creation with the Ingress Operator, there
+is a possibility of installing a version of the ExternalDNS Operator that lacks support for this same feature. However,
+it is important to note that the ExternalDNS Operator and the installer or Ingress Operator do not rely on each other's 
+logic. Hence, this mismatch does not pose a concern.
```

Comment:
Usually this section discusses version skew between the component and API server, or possibly version skew between the operator and its operand.  I suppose there are no such concerns here, however, as you are adding a new API field where EDNSO is the consumer, and the new functionality in EDNSO is setting a flag that already exists in the external-dns images that we have been shipping.